### ISSUE TITLE
misleading quota depletion errors when the build controller cannot create a builder pod

### ISSUE THREAD

**Issue by mbalazs** [openshift/origin#15877]
If a project has resource quotas set, and it contains an invalid buildConfig which causes the build controller to create an invalid builder pod, the user will see a misleading "Quota limit has been reached" error message.

This issue has come up together with #15876, but I'm creating a new issue because I believe this is an unrelated error that may also present in different circumstances. The steps to reproduce are the same as with #15876, but this time a resource quota must also be set in the project before deploying the application.

##### Version
oc v1.5.1
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://api.podspace.io:8443
openshift v1.5.1
kubernetes v1.5.2+43a9be4

##### Steps To Reproduce
1. Create a new project and set up the following quota and resource limits inside the project:

```
$ oc create -f - <<EOF
apiVersion: v1
items:
- apiVersion: v1
  kind: LimitRange
  metadata:
    creationTimestamp: null
    name: compute-resources-limitrange
  spec:
    limits:
    - default:
        cpu: "1"
        memory: 2Gi
      defaultRequest:
        cpu: 20m
        memory: 64Mi
      max:
        cpu: "4"
        memory: 8Gi
      min:
        cpu: 20m
        memory: 64Mi
      type: Container
kind: List
metadata: {}
EOF
```

```
$ oc create -f - <<EOF
apiVersion: v1
items:
- apiVersion: v1
  kind: ResourceQuota
  metadata:
    creationTimestamp: null
    name: compute-resources-quota
  spec:
    hard:
      limits.cpu: "6"
      limits.memory: 8Gi
      pods: "32"
    scopes:
    - NotTerminating
  status: {}
kind: List
metadata: {}
EOF
```

2. Follow the steps to reproduce  #15876 which will create a build config which fails to create a builder pod

##### Current Result
When the project is deployed after step 2., the user will immediately see a "Quota limit reached" error in the web console. The quota status page will show that the resource limit quotas are indeed depleted.

The project event log will contain these entries at this point:

```
LASTSEEN   FIRSTSEEN   COUNT     NAME      KIND      SUBOBJECT   TYPE      REASON             SOURCE                MESSAGE
2s         29m         28        hello-1   Build                 Warning   FailedCreate       {build-controller }   Error creating: Pod "hello-1-build" is invalid: [spec.volumes[2].name: Invalid value: "my.secret-source": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])? (e.g. 'my-name' or '123-abc'), spec.containers[0].volumeMounts[2].name: Not found: "my.secret-source"]
2s         29m         28        hello-1   Build                 Warning   HandleBuildError   {build-controller }   Build has error: failed to create build pod: Pod "hello-1-build" is invalid: [spec.volumes[2].name: Invalid value: "my.secret-source": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])? (e.g. 'my-name' or '123-abc'), spec.containers[0].volumeMounts[2].name: Not found: "my.secret-source"]
0s         29m         1739      hello-1   Build                 Warning   FailedCreate       {build-controller }   Error creating: pods "hello-1-build" is forbidden: exceeded quota: compute-resources-quota, requested: limits.memory=2Gi, used: limits.memory=8Gi, limited: limits.memory=8Gi
0s         29m         1739      hello-1   Build                 Warning   HandleBuildError   {build-controller }   Build has error: failed to create build pod: pods "hello-1-build" is forbidden: exceeded quota: compute-resources-quota, requested: limits.memory=2Gi, used: limits.memory=8Gi, limited: limits.memory=8Gi
```

##### Expected Result
The quota should not be depleted when the build controller fails to create the builder pod. No quota depletion error should be shown to the user, only the real cause of the failure (invalid volume name in this case)

**Comment by spadgett** [comment]
My understanding from @derekwaynecarr is that this is a known issue with quota. Invalid resources can temporarily count against quota, but it eventually corrects. The console is simply reporting what it sees in quota status, so I don't believe this is a console issue.

/unassign
/assign @sjenning
